core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -628,6 +628,24 @@ private[spark] class Executor(
           }
       } finally {
         runningTasks.remove(taskId)
+        if(conf.getBoolean("spark.conf.GCOnIdle.enabled", false)) {
+          if (runningTasks.isEmpty) {
+            triggerGCOnIdle(conf.getInt("spark.conf.GCOnIdle.threshold", 80))
+          }
+        }
+      }
+    }
+
+    /**
+     * Trigger GC on executor wait for remaining tasks to finish.
+     *
+     */
+    private def triggerGCOnIdle(threshold: Int): Unit = {
+      var memoryMXBean = ManagementFactory.getMemoryMXBean
```

Comment:
These should be `val`. The threshold should be a value between 0 and 1 too. The check can be
```
if (memoryMXBean.getHeapMemoryUsage.getUsed > threshold * memoryMXBean.getHeapMemoryUsage.getMax)
```